### ISSUE TITLE
Build failure: linuxPackages_6_10.sysdig

### ISSUE THREAD

**Issue by SebTM** [NixOS/nixpkgs#329233]
### Steps To Reproduce

1. build *sysdig*

### Build log

```
[ 63%] [MODERN BPF] Building BPF object: /build/source/build/driver/modern_bpf/vfork.bpf.o
clang: warning: ignoring '-fstack-protector-strong' option as it is not currently supported>
clang: warning: argument unused during compilation: '--gcc-toolchain=/nix/store/3s3rjkl3mx0>
[ 63%] [MODERN BPF] Building BPF object: /build/source/build/driver/modern_bpf/write.bpf.o
clang: warning: ignoring '-fstack-protector-strong' option as it is not currently supported>
clang: warning: argument unused during compilation: '--gcc-toolchain=/nix/store/3s3rjkl3mx0>
clang: warning: ignoring '-fstack-protector-strong' option as it is not currently supported>
clang: warning: argument unused during compilation: '--gcc-toolchain=/nix/store/3s3rjkl3mx0>
[ 63%] [MODERN BPF] Building BPF unique object file: /build/source/build/driver/modern_bpf/>
[ 63%] [MODERN BPF] Building BPF skeleton: /build/source/build/skel_dir/bpf_probe.skel.h
[ 63%] Built target ProbeSkeleton
[ 63%] Building C object libpman/CMakeFiles/pman.dir/src/stats.c.o
[ 63%] Building C object libpman/CMakeFiles/pman.dir/src/maps.c.o
[ 64%] Building C object libpman/CMakeFiles/pman.dir/src/lifecycle.c.o
[ 65%] Building C object libpman/CMakeFiles/pman.dir/src/configuration.c.o
[ 65%] Building C object libpman/CMakeFiles/pman.dir/src/ringbuffer.c.o
[ 65%] Building C object libpman/CMakeFiles/pman.dir/src/programs.c.o
[ 65%] Building C object libpman/CMakeFiles/pman.dir/src/sc_set.c.o
[ 65%] Building C object libpman/CMakeFiles/pman.dir/src/state.c.o
make[5]: *** [/nix/store/hlhp0pprnyq96npl2k1b5z9iy2xc4c9q-linux-6.10-dev/lib/modules/6.10.0>
make[4]: *** [/nix/store/hlhp0pprnyq96npl2k1b5z9iy2xc4c9q-linux-6.10-dev/lib/modules/6.10.0>
make[3]: *** [Makefile:22: all] Error 2
make[2]: *** [driver/CMakeFiles/driver.dir/build.make:70: driver/CMakeFiles/driver] Error 2
make[1]: *** [CMakeFiles/Makefile2:656: driver/CMakeFiles/driver.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 65%] Linking C static library libpman.a
[ 65%] Built target pman
[ 65%] Linking CXX static library libcri_v1alpha2.a
[ 65%] Built target cri_v1alpha2
[ 65%] Linking CXX static library libcri_v1.a
[ 65%] Built target cri_v1
make: *** [Makefile:156: all] Error 2
```

### Notify maintainers

@7c6f434c 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.9.9, NixOS, 24.11 (Vicuna), 24.11.20240716.b21b793`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.91.0-dev-pre20240717-6856720
System type: x86_64-linux
Additional system types: aarch64-linux, i686-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/sebtm/.config/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/sebtm/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/sebtm/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/sebtm/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf:/nix/store/y0996ncxalqaj216ngk9iwr5d3f6sdxp-gnome-settings-daemon-46.0/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/spcn1af2iapk3qaakv6dvnca1brs1mp4-lix-2.91.0-dev-pre20240717-6856720/share`
 - nixpkgs: `/nix/store/2c8gpdgjf3b0krn48k1hfcb340x72bcd-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by eclairevoyant** [comment]
Can't repro on master (09f65ccafc83ce77148375672be19e6746e45db8) and the error isn't in the log output, can you provide additional info?

**Comment by SebTM** [comment]
Hey, thanks for the response - checking again - it only happens if for me if I enable the sysdig-option in my config not building the sysdig-package - maybe this helps to reproduce and depends on the used kernel-version?

I've uploaded the full build log: https://gist.github.com/SebTM/73918e6d0d573fa5cb2bf03c033e908a

**Comment by eclairevoyant** [comment]
Thanks, seems this is the error:

```
/build/source/build/driver/src/main.c: In function 'drop_nostate_event':
/build/source/build/driver/src/main.c:1653:22: error: implicit declaration of function 'fd_is_open'; did you mean 'finish_open'? [-Werror=implicit-function-declaration]
 1653 |                     !fd_is_open(close_fd, fdt)
      |                      ^~~~~~~~~~
      |                      finish_open
```

**Comment by SebTM** [comment]
I guess this is already fixed upstream in the drivers (https://github.com/falcosecurity/libs/pull/1884) but needs to be added to nixpkgs to fix 6.10-kernel builds.

**Comment by SebTM** [comment]
https://github.com/NixOS/nixpkgs/pull/329418

fixes it for me
